packages/bs-neo3/src/services/vote/RpcVSNeo3.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  CalculateVoteFeeParams,
+  GenerateSigningCallbackResponse,
+  GetCandidatesToVoteResponse,
+  GetVoteDetailsByAddressResponse,
+  VoteParams,
+  VoteResponse,
+  VoteService,
+} from '../../interfaces'
+import { BSNeo3Helper } from '../../helpers/BSNeo3Helper'
+import { NeonInvoker } from '@cityofzion/neon-dappkit'
+import { Account, BSNumberHelper, Network, Token } from '@cityofzion/blockchain-service'
+import { ContractInvocationMulti } from '@cityofzion/neon-dappkit-types'
+import { tx } from '@cityofzion/neon-js'
+import { BSNeo3NetworkId } from '../../constants/BSNeo3Constants'
+
+type GetVoteCIMParams = {
+  address: string
+  candidatePubKey: string
+}
+
+type Params<BSName extends string> = {
+  network: Network<BSNeo3NetworkId>
+  feeToken: Token
+  generateSigningCallback: (account: Account<BSName>) => Promise<GenerateSigningCallbackResponse>
+}
+
+export class RpcVSNeo3<BSName extends string> implements VoteService<BSName> {
+  protected readonly network: Network<BSNeo3NetworkId>
+
+  readonly #feeToken: Token
+  readonly #generateSigningCallback: (account: Account<BSName>) => Promise<GenerateSigningCallbackResponse>
+
+  constructor({ network, feeToken, generateSigningCallback }: Params<BSName>) {
+    this.network = network
+    this.#feeToken = feeToken
+    this.#generateSigningCallback = generateSigningCallback
+  }
+
+  getCandidatesToVote(): Promise<GetCandidatesToVoteResponse> {
+    throw new Error('This method should be implemented')
+  }
+
+  getVoteDetailsByAddress(_address: string): Promise<GetVoteDetailsByAddressResponse> {
+    throw new Error('This method should be implemented')
+  }
```

Comment:
As this class cannot be used alone, it needs someone to extend it, it is better to use `abstract` instead of throwing an error.

```
abstract class RpcVoteServiceNeo3 {
  abstract getCandidatesToVote(): Promise<GetCandidatesToVoteResponse>

  abstract getVoteDetailsByAddress(_address: string): Promise<GetVoteDetailsByAddressResponse>
}
```

Reply:
I like it, but I can't test a abstract class, because it can't be instanced.